### PR TITLE
Add reward module endpoint - Closes #6689

### LINKED ISSUE #6689: Implement Reward Module endpoint
### Description
Implement `getDefaultRewardAtHeight` function as specified in https://research.lisk.com/t/define-state-transitions-of-reward-module/316#getdefaultrewardatheight-18 in `endpoint.ts` file and implement endpoint in reward module class.

### Acceptance Criteria
- `getDefaultRewardAtHeight` function is implemented and unit tested.

### Additional Information
- Reward brackets(milestones) are taken from `moduleConfig.brackets`
- `rewardOffset` and `rewardDistance` are taken from the config parameter in the `init` method of the reward class

### PR DESCRIPTION
**Pull request by mehmetegemen**
### What was the problem?

This PR resolves #6689

### How was it solved?

- Added `getDefaultRewardAtHeight` function and injected necessary configs for it

### How was it tested?

- With a unit test that tests every bracket and offset range
- With a unit test that tests module config is properly imported

### CODE REVIEW

File: framework/src/modules/reward/endpoint.ts [hunk 1]
```diff
@@ -12,6 +12,43 @@
...
+	private _offset!: number;
+	private _distance!: number;
+
+	public init(args: EndpointInitArgs) {
+		this._brackets = args.config.brackets;
+		this._offset = args.config.offset;
+		this._distance = args.config.distance;
+	}
+
+	public getDefaultRewardAtHeight(ctx: ModuleEndpointContext): DefaultReward {
```
Comment (shuse2) [inline framework/src/modules/reward/endpoint.ts:30]: `DefaultReward` has `bigint` in the property, which is not valid JSON format. It should be converted to `string`

File: framework/src/modules/reward/endpoint.ts [hunk 2]
```diff
@@ -12,6 +12,43 @@
...
+		const { height } = (ctx.params as unknown) as GetDefaultRewardParams;
```
Comment (shuse2) [inline framework/src/modules/reward/endpoint.ts:31]: Prefer to have validation than casting

**Review by shuse2 — CHANGES_REQUESTED** [review]

File: framework/src/modules/reward/endpoint.ts (hunk 2 above)
Comment (mehmetegemen) [reply framework/src/modules/reward/endpoint.ts]: done

File: framework/src/modules/reward/endpoint.ts (hunk 1 above)
Comment (mehmetegemen) [reply framework/src/modules/reward/endpoint.ts:30]: done

**Review by shuse2 — APPROVED** [review]

File: framework/src/modules/reward/endpoint.ts [hunk 3]
```diff
@@ -12,6 +12,51 @@
...
+		this._brackets = args.config.brackets;
+		this._offset = args.config.offset;
+		this._distance = args.config.distance;
+	}
+
+	public getDefaultRewardAtHeight(ctx: ModuleEndpointContext): DefaultReward {
+		const { height } = ctx.params;
+
+		if (typeof height !== 'number') {
+			throw new Error('Parameter height must be a number');
```
Comment (ishantiw) [inline framework/src/modules/reward/endpoint.ts:34]:
```suggestion
			throw new Error('Parameter height must be a number.');
```

File: framework/src/modules/reward/endpoint.ts [hunk 4]
```diff
@@ -12,6 +12,51 @@
...
+		}
+
+		if (height < 0) {
+			throw new Error('Parameter height cannot be smaller than 0');
```
Comment (ishantiw) [inline framework/src/modules/reward/endpoint.ts:38]:
```suggestion
			throw new Error('Parameter height cannot be smaller than 0.');
```

File: framework/src/modules/reward/endpoint.ts [hunk 5]
```diff
@@ -12,6 +12,51 @@
...
+		const distance = Math.floor(this._distance);
+		const location = Math.trunc((height - this._offset) / distance);
+		const lastBracket = this._brackets[this._brackets.length - 1];
+
+		let bracket;
+		if (location > this._brackets.length - 1) {
+			bracket = this._brackets.lastIndexOf(lastBracket);
+		} else {
+			bracket = location;
+		}
```
Comment (ishantiw) [inline framework/src/modules/reward/endpoint.ts:56]:
```suggestion
		const bracket = (location > this._brackets.length - 1) ? this._brackets.lastIndexOf(lastBracket) : location;
```

File: framework/test/unit/modules/reward/endpoint.spec.ts [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+					logger,
+					params: {
+						height: currentHeight,
+					},
+				});
+				expect(rewardFromEndpoint).toEqual({ reward: rewardFromConfig.toString() });
+			});
+		}
+
+		it(`should getDefaultRewardAtHeight work for the height below offset`, () => {
```
Comment (ishantiw) [inline framework/test/unit/modules/reward/endpoint.spec.ts:70]:
```suggestion
		it('should getDefaultRewardAtHeight work for the height below offset', () => {
```

File: framework/test/unit/modules/reward/endpoint.spec.ts [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+import { Logger } from '../../../../src/logger';
+import { RewardModule } from '../../../../src/modules/reward';
+import { fakeLogger } from '../../../utils/node';
+
+describe('RewardModule', () => {
```
Comment (ishantiw) [inline framework/test/unit/modules/reward/endpoint.spec.ts:18]:
```suggestion
describe('RewardModuleEndpoint', () => {
```

`reward_module.spec.ts` also have the same test suite name, also maybe we don't need another describe inside as `endpoint`

**Review by ishantiw — CHANGES_REQUESTED** [review]

File: framework/test/unit/modules/reward/endpoint.spec.ts (hunk 7 above)
Comment (mehmetegemen) [reply framework/test/unit/modules/reward/endpoint.spec.ts:18]: you are correct, there is a redundancy

File: framework/src/modules/reward/endpoint.ts (hunk 5 above)
Comment (mehmetegemen) [reply framework/src/modules/reward/endpoint.ts:56]: very nice suggestion, will implement

File: framework/src/modules/reward/endpoint.ts (hunk 3 above)
Comment (mehmetegemen) [reply framework/src/modules/reward/endpoint.ts:34]: I couldn't remember our convention for error messages, I'll add the dots as suggested

Comment (mehmetegemen) [reply framework/src/modules/reward/endpoint.ts:34]: done

File: framework/src/modules/reward/endpoint.ts (hunk 4 above)
Comment (mehmetegemen) [reply framework/src/modules/reward/endpoint.ts:38]: done

File: framework/src/modules/reward/endpoint.ts (hunk 5 above)
Comment (mehmetegemen) [reply framework/src/modules/reward/endpoint.ts:56]: done

File: framework/test/unit/modules/reward/endpoint.spec.ts (hunk 7 above)
Comment (mehmetegemen) [reply framework/test/unit/modules/reward/endpoint.spec.ts:18]: done

File: framework/test/unit/modules/reward/endpoint.spec.ts (hunk 6 above)
Comment (mehmetegemen) [reply framework/test/unit/modules/reward/endpoint.spec.ts:70]: done

File: framework/src/modules/reward/calculate_reward.ts [hunk 8]
```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright © 2020 Lisk Foundation
```
Comment (ishantiw) [inline framework/src/modules/reward/calculate_reward.ts:2]:
```suggestion
 * Copyright © 2021 Lisk Foundation
```
Ignore this comment if this file was only moved

**Review by ishantiw — APPROVED** [review]
LGTM 🚀

File: framework/src/modules/reward/calculate_reward.ts (hunk 8 above)
Comment (mehmetegemen) [reply framework/src/modules/reward/calculate_reward.ts:2]: It wasn't moved, need to fix it 😄

**Review by shuse2 — APPROVED** [review]
Other than @ishantiw 's comment LGTM

File: framework/src/modules/reward/calculate_reward.ts (hunk 8 above)
Comment (mehmetegemen) [reply framework/src/modules/reward/calculate_reward.ts]: done

**Pull request merged**